sdk/python/kfp/v2/components/executor.py

```diff
@@ -251,13 +251,12 @@ def _write_executor_output(self, func_output: Optional[Any] = None):
                     'Unknown return type: {}. Must be one of `str`, `int`, `float`, a'
                     ' subclass of `Artifact`, or a NamedTuple collection of these types.'
                     .format(self._return_annotation))
-
         import os
-        os.makedirs(
-            os.path.dirname(self._input['outputs']['outputFile']),
-            exist_ok=True)
-        with open(self._input['outputs']['outputFile'], 'w') as f:
-            f.write(json.dumps(self._executor_output))
+        executor_output_path = self._input['outputs']['outputFile']
+        if not os.path.exists(executor_output_path):
```

Comment:
As discussed offline, an `if` check would reduce the chance of race condition when this code is run by multiple workers at the same time, but it doesn't solve the problem completely, we should push for the underlying GCS fuse fix while going forward with this mitigation solution.
Secondly, we should have some comments above, without the context, the change doesn't make sense from a pure Python code review perspective, the next engineer come in without the context may possible revert the change.